### DISCUSSION TITLE
[back] socket 에서 bad reqquest exception 발생시 처리

### DISCUSSION THREAD

**Discussion by kyu-baek** [3DPong/transcendence#202] [Q&A]
gateway 에서 
@UsePipes(new ValidationPipe()) 를 사용하면, 
파이프에서 에러가 발생할 때 아래와 같이 에러가 처리됩니다.

![image](https://user-images.githubusercontent.com/69143394/234513335-5ffdc688-a775-4f49-bf26-1bae89c68e24.png)

이런 에러가 발생했을 nestjs 의 공식 문서에서는 

https://docs.nestjs.com/websockets/pipes

WsException 을 통해 관리 하라고 하는데, 이를 적용하는 게 어렵습니다.
제가 생각하는 해결 방식은

1. bad reqquest exception 이 발생했을 때 자동으로 필터로 넘어간다
2. 해당 필터에서는 상황에 맞는 message를 throw new SocketException('BadRequest', `500자 이하`); 으로 보내준다.
3. 프론트에서 error 이벤트로 문제를 클라이언트에게 띄운다 

인데 3번은 이미 자동화 되어 있는 상태고, 1,2번을 만드는 게 막힌 상태입니다.

**Comment by JuneParkCode** [comment]
삭제 (잘못알고 있었네요)
--

`WsException 을 통해 관리 하라고 하는데, 이를 적용하는 게 어렵습니다.` 정확히 어떤 부분이 어려우신건가요?  Validation Pipe 문서(올려두신거)를 읽어보니 WsException을 throw 한다고 하니, 그러면 핸들링 가능한 것 같은데...

**Reply by kyu-baek** [reply]
handleChatEvent(@ConnectedSocket() socket: Socket, @MessageBody() md: MessageDto)  여기처럼 MessageDto 에서 사용하는 class-validator 가 WsException 이 아닌 badRequest Exception을 발생시키기 때문입니다.

그래서 badRequestException -> WsException 으로 변환하는 코드를 짰는데, 

결과적으로 저렇게 인자로 함수 호출과 동시에 받아질 때는 함수 내부의 try catch 문으로 가지 않아서 MessageDto 검증은 일단 가져와서 함수 내부에서 할까 고민중입니다.

**Reply by JuneParkCode** [reply]
Filter 자체에서 HttpException을 받을 수 있게 하면 되지 랂을까요
